src/common/logger/logger.cpp

```diff
@@ -33,31 +33,25 @@ level::level_enum getLogLevel(std::wstring_view logSettingsPath)
     return result;
 }
 
-Logger::Logger()
-{
-}
+std::shared_ptr<spdlog::logger> Logger::logger;
 
-Logger::Logger(std::string loggerName, std::wstring logFilePath, std::wstring_view logSettingsPath)
+void Logger::init(std::string loggerName, std::wstring logFilePath, std::wstring_view logSettingsPath)
 {
     auto logLevel = getLogLevel(logSettingsPath);
     try
     {
         auto sink = make_shared<sinks::daily_file_sink_mt>(logFilePath, 0, 0, false, LogSettings::retention);
-        this->logger = make_shared<spdlog::logger>(loggerName, sink);
+        logger = make_shared<spdlog::logger>(loggerName, sink);
     }
     catch (...)
     {
         cerr << "Can not create file logger. Create stdout logger instead" << endl;
-        this->logger = spdlog::stdout_color_mt("some_unique_name");
+        logger = spdlog::stdout_color_mt("some_unique_name");
```

Comment:
How is this supposed to work?

Reply:
If a logger can not be created then we create a logger that writes to stdout

Reply:
It is important that a logger is not null so it will not crash on logging

Reply:
Changed name of the logger